server/api/resolvers/clientResolvers.ts

```diff
@@ -0,0 +1,45 @@
+import { Client, ClientInterface } from '../../database/models/clientModel'
+
+const clientQueryResolvers = {
+    client: async (_, { _id }, { authenticateUser }): Promise<ClientInterface> => {
+        return authenticateUser().then(() => {
+            return Client.findById(_id).exec()
+        }) 
+
+    },
+    clients: async (_, __, { authenticateUser }): Promise<Array<ClientInterface>> => {
+        return authenticateUser().then(() => {
+            return Client.find().exec()
+        })
+    },
+    clientsFilter: async (_, { filter, options }, { authenticateUser }): Promise<Array<ClientInterface>> => {
+        return authenticateUser().then(() => {
+            return Client.find({ fullName: filter.fullName }).exec()
+        })
+    },
+}
+
+const clientMutationResolvers = {
+    createClient: async (_, { client }, { authenticateUser }): Promise<ClientInterface> => {
+        return authenticateUser().then(async () => {
+            const newClient = new Client({...client})
+            return newClient.save()
+        })
+    },
+    updateClient: async (_, { client }, { authenticateUser }): Promise<ClientInterface> => {
+        return authenticateUser().then(async () => {
+            const currentClient = await Client.findById(client._id)
```

Comment:
Should we verify the client object contains the field _id? Same with the other endpoints.

Reply:
In the graphql schema, _id is set as mandatory